src/Producer/RecordValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+
+namespace Kafka\Producer;
+
+use Kafka\Broker;
+use Kafka\Exception;
+use function is_string;
+use function trim;
+
+class RecordValidator
+{
+    /**
+     * @var string[][]
+     */
+    private $topics;
+
+    public function __construct(Broker $broker)
+    {
+        $this->topics = $broker->getTopics();
+    }
+
+    /**
+     * @param string[] $record
+     */
+    public function validate(array $record): void
```

Comment:
We could send the topic list as an argument here, then we completely remove the `Broker` dependency.